src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -306,21 +306,26 @@ public bool IsSolutionOpen
             {
                 return NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
                 {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                    var dte = await _asyncServiceProvider.GetDTEAsync();
-                    return dte != null &&
-                           dte.Solution != null &&
-                           dte.Solution.IsOpen;
+                    return await IsSolutionOpenAsync();
                 });
             }
         }
 
+        public async Task<bool> IsSolutionOpenAsync()
```

Comment:
I found another implementation to check if solution is open in VS SDK Extensiblility samples repo. I believe the difference is [this](https://github.com/microsoft/VSSDK-Extensibility-Samples/blob/e6e86e6405095de060cbf3838ce7df2cd28ca0ae/SolutionLoadEvents/src/VSPackage.cs#L35) implementation avoids using `EnvDTE`

Reply:
The DTE calls IVSSolution under neat the hood.
I think the guidance in general is to not use the DTE, so should maybe create a task to move to IVsSolution instead.

Reply:
Why not use the code the code that Kartheek linked? The method signatures are identical, and we can avoid putting this off for later.

Reply:
Discussed offline with @zivkan.

My opinion is that we should not make method by method increments but rather address the general change in the whole class. 

Do we have another motivation beyond don't use the DTE? 
If there's a tangible improvement, let's do that in a different work item rather than introduce the usage of different services in the same class.